File: src/pcb_ai/schemas/knowledge.py
"""Engineering knowledge chunk schema for the RAG layer (Phase 6)."""
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field

KNOWLEDGE_SCHEMA_VERSION = "1.0"


class KnowledgeChunk(BaseModel):
    schema_version: str = KNOWLEDGE_SCHEMA_VERSION
    document_id: str
    document_name: str
    source: str  # datasheet | design_guideline | app_note | internal_doc | ipc | other
    page: Optional[int] = None
    section: Optional[str] = None
    chunk_id: str
    text: str
    component_references: list[str] = Field(default_factory=list)
    keywords: list[str] = Field(default_factory=list)
    similarity: Optional[float] = None

File: frontend/src/types/knowledge.ts
export interface KnowledgeShowcaseDocument {
  document_id: string
  document_name: string
  label: string
  project: string
  source: string
  chunk_count: number
  page_count: number | null
}

export interface KnowledgeShowcaseSummary {
  document_count: number
  total_chunks: number
  documents: KnowledgeShowcaseDocument[]
}

export interface KnowledgeChunkResult {
  schema_version: string
  document_id: string
  document_name: string
  source: string
  page: number | null
  section: string | null
  chunk_id: string
  text: string
  component_references: string[]
  keywords: string[]
  similarity: number | null
}

export interface KnowledgeSearchResponse {
  query: string
  results: KnowledgeChunkResult[]
}
